### DISCUSSION TITLE
Dishwasher as valce

### DISCUSSION THREAD

**Discussion by ringostarlet** [thoukydides/homebridge-homeconnect#218] [Ideas]
Hi, I have a Miele washing machine and it shows in Homekit as a valve, including remaining time and all. Its somewhat more intuitive than just switches, could the way dishwashers are displayed in Homeconnect also be valves?

**Comment by thoukydides** [comment]
This has been discussed previously. See #48, #68, #114, etc…

Could it be done? Yes.
Will it be done? No. 

My primary objection is that it breaks the expected/intended semantics. This tends to cause problems with Siri control. Whilst there might be some association between irrigation systems and a dishwasher / washing machine, that really doesn’t apply for the many other Home Connect appliance types supported by this plugin. This would make is less intuitive rather than more.

HomeKit (and the Apple Home app in particular) is never going to be a good way to view or control appliances like this that do not fit its object model. As far as I am concerned, the only reasons to bother integrating Home Connect with HomeKit are (i) to allow control from Siri and (ii) to trigger automations. For everything else it is better to just use the actual Home Connect app. Implementing less appropriate services will make the first of those worse and won’t help the second.

I know other people have different opinions… and they are welcome to develop their own alternative Homebridge plugins (or fork this one) to meet their own requirements.
